### PR TITLE
Create add_driver_inf cmake function

### PR DESCRIPTION
**Pull request by Extravert-ir**
*.inf files for drivers can now be placed along with the driver code

Now sure what to do with inf files, dealing with several drivers at once

### CODE REVIEW

File: drivers/processor/processr/CMakeLists.txt [hunk 1]
```diff
@@ -10,3 +10,4 @@ set_module_type(processr kernelmodedriver)
 add_importlibs(processr hal ntoskrnl)
 add_pch(processr processr.h SOURCE)
 add_cd_file(TARGET processr DESTINATION reactos/system32/drivers FOR all)
+add_driver_inf(processr processr.inf)
```
Comment (HBelusca) [inline drivers/processor/processr/CMakeLists.txt]: The standard name of the inf file stored within %SystemRoot%/inf is "cpu.inf" on Windows.

File: drivers/usb/usbhub/usbhub.inf [hunk 2]
```diff
@@ -1,4 +1,6 @@
-;USB.INF
+; usbhub.sys
```
Comment (HBelusca) [inline drivers/usb/usbhub/usbhub.inf]: Please don't!! This is breaking WIndows compatibility.
When you have a INF file that covers "many" drivers, this means that this set of drivers are part of "one" module: here for example the set of USB drivers.
So instead you should keep the whole USB.INF file as it is and add the "add_driver_inf" command in the main drivers/usb/ cmakelists file.

File: drivers/processor/processr/CMakeLists.txt (hunk 1 above)
Comment (ThFabba) [reply drivers/processor/processr/CMakeLists.txt]: This applies to a bunch of the others too. There's probably value in keeping the same inf names as Windows, though I couldn't say for sure.

File: sdk/cmake/CMakeMacros.cmake [hunk 3]
```diff
@@ -908,6 +908,21 @@ function(create_registry_hives)
 
 endfunction()
 
+function(add_driver_inf _module)
```
Comment (ThFabba) [inline sdk/cmake/CMakeMacros.cmake]: Would be nice if this function were designed such that `media/inf/CMakeLists.txt` can use it, too.

File: media/inf/CMakeLists.txt [hunk 4]
```diff
@@ -25,22 +19,13 @@ list(APPEND INF_FILES
...
     ports.inf
     scsi.inf
     shortcuts.inf
     unknown.inf
-    usb.inf
-    usbport.inf
-    usbstor.inf
     wdmaudio.inf
-    wmiacpi.inf
-    xboxdisp.inf)
```
Comment (ThFabba) [inline media/inf/CMakeLists.txt]: You moved this but don't seem to have added it to another CMake file.

File: drivers/usb/usbhub/usbhub.inf (hunk 2 above)
Comment (Extravert-ir) [reply drivers/usb/usbhub/usbhub.inf:1]: The current separation to `usb.inf`, `usbport.inf` and `usbstor.inf` does not make much sense either - these drivers don't work without each other.

File: drivers/processor/processr/CMakeLists.txt (hunk 1 above)
Comment (Extravert-ir) [reply drivers/processor/processr/CMakeLists.txt:13]: I tried, but can't find what can be the reason to keep inf file names the same. Can you help? :)

Comment (HBelusca) [reply drivers/processor/processr/CMakeLists.txt:13]: A 3rd-party driver's INF file could "import" an existing Windows INF file via the "LayoutFile" directive to base itself upon.

Comment (Extravert-ir) [reply drivers/processor/processr/CMakeLists.txt:13]: So what do you suggest regarding `usb.inf` and friends - leave them as it is in `media/inf` folder?

**Comment by HBelusca** [comment]
The idea of the DriverVer automatic generation using a cmake script, but for maximum Windows compatibility we shouldn't use our own version (e.g. 0.4.x-yyyy) but instead the corresponding Windows version we are compatible with, which is defined in
https://git.reactos.org/?p=reactos.git;a=blob;f=sdk/include/psdk/ntverp.h;h=a3fd98808d1c470ba6d27226b9afa36341b53d8e;hb=HEAD

File: drivers/storage/class/disk/CMakeLists.txt [hunk 5]
```diff
@@ -5,3 +5,4 @@ set_module_type(disk kernelmodedriver)
 add_importlibs(disk class2 scsiport ntoskrnl hal)
 add_cd_file(TARGET disk DESTINATION reactos/system32/drivers NO_CAB FOR all)
 add_registry_inf(disk_reg.inf)
+add_driver_inf(disk disk.inf)
```
Comment (ThFabba) [inline drivers/storage/class/disk/CMakeLists.txt:8]: Maybe add a commented version of this to disk_new to remind people once this gets switched.

**Pull request merged**